awx/api/serializers.py

```diff
@@ -4419,6 +4555,333 @@ def validate(self, attrs):
+        if 'organization' not in attrs or attrs['organization'] == None or attrs['organization'] == '':
+            if Organization.accessible_pk_qs(request.user, 'read_role').count() == 1:
+                for tup in Organization.accessible_pk_qs(request.user, 'read_role').all():
+                    attrs['organization'] = Organization.objects.filter(id__in=str(tup[0])).first()
+            elif Organization.accessible_pk_qs(request.user, 'read_role').count() > 1:
+                raise serializers.ValidationError("User has permission to multiple Organizations, please set one of them in the request")
+            else:
+                raise serializers.ValidationError("User not part of any organization, please assign an organization to assign to the bulk job")
+        else:
+            allowed_orgs = set()
+            requested_org = attrs['organization']
+            if request and not request.user.is_superuser:
+                [allowed_orgs.add(tup[0]) for tup in Organization.accessible_pk_qs(request.user, 'read_role').all()]
+                if requested_org.id not in allowed_orgs:
+                    raise ValidationError(_(f"Organization {requested_org.id} not found"))
+
+    def check_unified_job_permission(self, request, requested_ujts):
+        allowed_ujts = set()
+        [allowed_ujts.add(tup[0]) for tup in UnifiedJobTemplate.accessible_pk_qs(request.user, 'execute_role').all()]
+        [allowed_ujts.add(tup[0]) for tup in UnifiedJobTemplate.accessible_pk_qs(request.user, 'admin_role').all()]
+        [allowed_ujts.add(tup[0]) for tup in UnifiedJobTemplate.accessible_pk_qs(request.user, 'update_role').all()]
+        accessible_inventories_qs = Inventory.accessible_pk_qs(request.user, 'update_role')
+        [allowed_ujts.add(tup[0]) for tup in InventorySource.objects.filter(inventory__in=accessible_inventories_qs).values_list('id')]
+
+        if requested_ujts - allowed_ujts:
+            not_allowed = requested_ujts - allowed_ujts
+            raise serializers.ValidationError(_(f"Unified Job Templates {not_allowed} not found."))
+
+    def check_inventory_permission(self, request, requested_use_inventories):
+        accessible_use_inventories = {tup[0] for tup in Inventory.accessible_pk_qs(request.user, 'use_role')}
+        if requested_use_inventories - accessible_use_inventories:
+            not_allowed = requested_use_inventories - accessible_use_inventories
+            raise serializers.ValidationError(_(f"Inventories {not_allowed} not found."))
+
+    def check_credential_permission(self, request, requested_use_credentials):
+        accessible_use_credentials = {tup[0] for tup in Credential.accessible_pk_qs(request.user, 'use_role').all()}
+        if requested_use_credentials - accessible_use_credentials:
+            not_allowed = requested_use_credentials - accessible_use_credentials
+            raise serializers.ValidationError(_(f"Credentials {not_allowed} not found."))
```

Comment:
they may be viable - the user doesn't have use role for them. Maybe say "not found or you don't have permission" because that's pretty normal to avoid divulging info the user shouldn't have.

Reply:
Sure. Done